Source/Classes/FrameStore.swift

```diff
@@ -177,6 +190,11 @@ private extension FrameStore {
   /// Increments the `currentFrameIndex` property.
   func incrementCurrentFrameIndex() {
     currentFrameIndex = increment(frameIndex: currentFrameIndex)
+    if currentFrameIndex == frameCount - 1 && currentLoop == loopCount - 1 {
```

Comment:
I'd extract these conditions into 2 computed properties `isLastFrame` and `isLastLoop`